clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -514,7 +514,7 @@ private TransactionManager configureTransactionState(ProducerConfig config,
                                                          LogContext logContext) {
         TransactionManager transactionManager = null;
 
-        if (config.idempotenceEnabled()) {
+        if (config.getBoolean(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG)) {
```

Comment:
refactor: the `idempotenceEnabled()` method is only to get the config of `ENABLE_IDEMPOTENCE_CONFIG` now.